packages/cli/src/models/network/NetworkController.ts

```diff
@@ -464,40 +463,34 @@ export default class NetworkController {
   }
 
   // Contract model || SolidityLib model
-  private _hasSolidityLibChanged(libClass: Contract): boolean {
+  private hasSolidityLibChanged(libClass: Contract): boolean {
     return !this.networkFile.hasSameBytecode(libClass.schema.contractName, libClass);
   }
 
   // Contract model
-  public hasContractChanged(contractAlias: string, contract?: Contract): boolean {
-    if (!this.isLocalContract(contractAlias)) return false;
-    if (!this.isContractDeployed(contractAlias)) return true;
+  public hasContractChanged(contractName: string, contract?: Contract): boolean {
+    if (!this.isProjectFileContract(contractName)) return false;
+    if (this.isProjectFileContract(contractName) && !this.isNetworkFileContract(contractName)) return true;
```

Comment:
The first part of the second condition is implied by the previous line.

```suggestion
    if (!this.isProjectFileContract(contractName)) return false;
    if (!this.isNetworkFileContract(contractName)) return true;
```

It may be clearer using an `if/else`.
```suggestion
    if (!this.isProjectFileContract(contractName)) {
      return false;
    } else if (!this.isNetworkFileContract(contractName)) {
      return true;
    }
```